### PR TITLE
EditorWithBufferGeometry

### PR DESCRIPTION
**Pull request by rfm1201**
When you add a geometry you can choose normal or buffered geometry.
When you update the geometry, it stay buffered or not buffered depending of its nature.
Should not be in a release but can be useful when validating that both kind of geometry work the same way

### CODE REVIEW

**Comment by tschw** [comment]
IIRC there's a "Convert to BufferGeometry" button, maybe that would should be made aware of the custom Geometries? Just a thought...

File: editor/js/Menubar.Add.js [hunk 1]
```diff
@@ -54,12 +74,12 @@ Menubar.Add = function ( editor ) {
 	var option = new UI.Row();
 	option.setClass( 'option' );
 	option.setTextContent( 'Plane' );
-	option.onClick( function () {
+	option.onClick( function() {
 
-		var geometry = new THREE.PlaneGeometry( 2, 2 );
+		var geometry = new THREE[ buffered ? "PlaneBufferGeometry" : "PlaneGeometry" ]( 2, 2 );
```
Comment (mrdoob) [inline editor/js/Menubar.Add.js]: At this point, I think I would just change them to use BufferGeometries directly (without the checkbox).

**Comment by rfm1201** [comment]
I put a checkbox in my own version to be able to compare the two versions (especially that I have no problem with UVs).
I didn't found any problem with the buffered versions, so the option without checkbox would be ok from a user point of view.
If all the standard geometries used in the editor became buffered one (there's one missing but as I don't have the code with me now, I forget witch one), the easiest way is to rename all the Sidebar.Geometry.XxxGeometry stuff and not to use this pull request.

**Comment by tschw** [comment]
Are there actually features that the unbuffered geometries have over the buffered ones?

Seeing #8165, I wonder whether there should just be a buffer-based `EditMesh` kind of thing to replace the `Geometry` base class...

**Comment by mrdoob** [comment]
> Are there actually features that the unbuffered geometries have over the buffered ones?

The only benefit of `Geometry` is that vertices, normals, ... do not need to share the same indices.

Having said that, I was hoping to eventually replace `Geometry` with `BufferGeometry`. `Geometry` becoming `LegacyGeometry`. @Mugen87 has done a great job lately porting primitives to `BufferGeometry`. There are still a few things that rely on `Geometry` though (we should probably make a list).

**Comment by mrdoob** [comment]
Implemented with the checkbox and without the equivalence object.

**Comment by mrdoob** [comment]
Thanks!

**Pull request closed without merging**